src/devices_models/device_constructors/thermalgeneration_constructor.jl

```diff
@@ -802,13 +783,9 @@ function construct_device!(
     container::OptimizationContainer,
     sys::PSY.System,
     ::ModelConstructStage,
-    model::DeviceModel{T, D},
-    network_model::NetworkModel{S},
-) where {
-    T <: PSY.ThermalGen,
-    D <: AbstractThermalDispatchFormulation,
-    S <: PM.AbstractActivePowerModel,
-}
+    model::DeviceModel{T, <:AbstractThermalDispatchFormulation},
```

Comment:
Changes to DeviceModel were not required. However, since these type variables were not used, I removed them.

Reply:
That's ok